python/ray/tests/test_basic_5.py

```diff
@@ -119,5 +120,24 @@ def test_internal_kv(ray_start_regular):
         kv._internal_kv_list("@namespace_abc", namespace="n")
 
 
+@pytest.mark.skipif(
+    os.environ.get("CI") and sys.platform != "linux",
+    reason="This test is only run on linux CI machines.")
+def test_using_hostname(ray_start_cluster):
+    import socket
+    hostname = socket.gethostname()
+    cluster = Cluster(
+        initialize_head=True, head_node_args={
+            "node_ip_address": hostname,
+        })
+
+    # Use `ray.init` to test the connection.
+    ray.init(address=cluster.address, _node_ip_address=hostname)
```

Comment:
Let me make it clear. If we use IP addresses everywhere, we don't need to provide the `_node_ip_address` argument when starting a driver script, but it's needed if we use hostnames everywhere, right? Is there any way to omit `_node_ip_address` when hostnames are used?

Reply:
Actually, in my one node test, the `_node_ip_address` can be omitted.

Reply:
Then maybe we should remove `_node_ip_address` from the test case?

Reply:
I don't think we should remove `_node_ip_address`.  I think there are some special optimization for the case in which address is found automatically. Maybe it couldn't work in a multiple nodes cluster(I just guess). And here we must ensure  `_node_ip_address=hostname` could work.